### PR TITLE
optionally enable build failures caused by linting

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- introduce variable to allow failing builds on linting issues
- fail eventing-controller build on linting issues

**Related issue(s)**

#12468

### CODE REVIEW

**Review by koala7659 — DISMISSED** [review]
LGTM

File: common/makefiles/generic-make-go.mk [hunk 1]
```diff
@@ -186,7 +191,7 @@ vet-local:
 
 # Lint goal is optional for now. remove the - at the beginning to make the goal fail if there are linting errors
```
Comment (radufa) [inline common/makefiles/generic-make-go.mk:192]: Perhaps this comment should be changed too.

**Review by radufa — APPROVED** [review]

**Pull request merged**